altair/api.py

```diff
@@ -409,6 +409,70 @@ def to_dict(self):
         D['config'] = self.vlconfig.to_dict()
         return D
 
+    def select_x(self):
+        """
+        Helper function that does a best effort of selecting an automatic x axis.
+
+        Raises AssertionError if it cannot find a x axis to choose.
+        """
+        assert self.data is not None
+        assert len(self.data) >= 1
+
+        d = self._classify_data_by_type()
+
+        # Choose the first column found on the following order: T, O, N, Q
+        chosen_x = None
+        for typ in ['Q', 'N', 'O', 'T']:
+            if len(d[typ]) >= 1:
+                chosen_x = d[typ][0]
+
+        if chosen_x is None:
+            raise AssertionError('Could not select X.')
```

Comment:
I suppose either is fine – I've just never seen an AssertionError called directly (as opposed to, e.g. ``assert <statement>``) so it struck me as odd.

On the loop thing – either is fine. It just took me a few reads to figure out why the comment didn't match the code at first glance.